### PR TITLE
Fix formatFields in all shops context

### PR DESCRIPTION
**Pull request by jrruiz**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix a bug when comparing the shop property with the value 'both', when the field has the shop property with the value true.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | In a shop with the multishop feature active. Set a product with different price in two shops. Change the context to all shops and save the product. When you save the product with the all shops context, the price is changed in every shop although the multishop_check for the price is unchecked.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/7624)

### CODE REVIEW

**Review by atomiix — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Just to be sure behat tests are not failing for nothing :sweat_smile:

**Review by PierreRambaud — APPROVED** [review]
Doesn't require QA. Can be approved and merge by maintainers

**Review by matthieu-rolland — COMMENTED** [review]
Seems like this modification breaks behat test ? 
I restart it, just in case since I'm not sure about the error message but...

**Comment by PierreRambaud** [comment]
> Seems like this modification breaks behat test ?
> I restart it, just in case since I'm not sure about the error message but...

How is it possible oO, should this PR needs a rebase?

**Review by matthieu-rolland — COMMENTED** [review]
Yep, seems like there's a behat test that need some fixing before this PR can be merged. 
Do you want to look into it @jrruiz  ?

**Comment by PierreRambaud** [comment]
Hi @jrruiz ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**
